### PR TITLE
front: fix power restrictions warnings

### LINKED ISSUE #6543: Double warning message for power restrictions
### What happened?

The following message is displayed twice : 

Image

### What did you expect to happen?

The messages shouldn't be displayed as a toast anymore as the user might miss them or might don't have the time the read it before it vanishes. They should rather be static and close to the editor.

Ideally, the invalidty warnings should be displayed first, before the missing code warning.

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to the train creation
2. Try to select power restriction 
3. If no restriction is already selected the 2 messages appear 

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: Windows
- **Env**: Recette

### OSRD version (top right corner `Account` button > `Informations`)

staging e220992

### PR DESCRIPTION
**Pull request by SharglutDev**
close #6543

### CODE REVIEW

File: front/public/locales/fr/operationalStudies/manageTrainSchedule.json [hunk 1]
```diff
@@ -115,8 +115,10 @@
   "vias": "Points de passage, arrêts",
   "warningMessages": {
     "electrification": "Électrification",
-    "powerRestrictionInvalidCombination": "Code {{powerRestrictionCode}} incompatible avec l'électrification à {{electrification}} de l'itinéraire entre {{begin}}m et {{end}}m. ",
-    "missingPowerRestriction": "Restriction de puissance manquante entre {{begin}}m et {{end}}m.",
+    "inconsistent_one": "Un intervalle comporte des incohérences. La courbe effort vitesse par défaut sera utilisée pour celui-ci.",
+    "inconsistent_many": "Des intervalles comportent des incohérences. Les courbes effort vitesse par défaut seront utilisées pour ceux-ci.",
```
Comment (clarani) [inline front/public/locales/fr/operationalStudies/manageTrainSchedule.json]:
```suggestion
    "inconsistent_many": "Plusieurs intervalles comportent des incohérences. Les courbes effort vitesse par défaut seront utilisées pour ceux-ci.",
```

File: front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx [hunk 2]
```diff
@@ -203,6 +140,39 @@ const PowerRestrictionsSelector = ({
...
           <>
             <p className="mb-1 mt-1">{t('powerRestrictionExplanationText')}</p>
+            {totalPowerRestrictionWarnings > 0 && (
+              <div className="border border-warning rounded p-3 my-3">
+                <div className="d-flex align-items-center mb-2">
+                  <Alert />
+                  <span className="ml-2">
+                    {totalPowerRestrictionWarnings > 1
+                      ? t('warningMessages.inconsistent_many')
+                      : t('warningMessages.inconsistent_one')}
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: i18n can handle pluralize, you can have a look at infrasFound_other in the translation files :)
```suggestion
                    {t('warningMessages.inconsistent', {count: totalPowerRestrictionWarnings})}
```

File: front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx [hunk 3]
```diff
@@ -203,6 +140,39 @@ const PowerRestrictionsSelector = ({
...
+                  </span>
+                </div>
+                {Object.keys(powerRestrictionsWarnings).map((powerRestrictionCategory) => (
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: You can use Object.entries to directly get the key and value:
```suggestion
                {Object.entries(powerRestrictionsWarnings).map(
                  ([warningCategory, warningsByCategory]) => 
```

**Review by achrafmohye — APPROVED** [review]
looks good !

File: front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export type PowerRestrictionWarning = {
+  powerRestrictionCode?: string;
+  electrification?: string;
+  begin: number;
+  end: number;
+};
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts:6]: You can make the type more precise
```suggestion
export type PowerRestrictionWarning = {
  begin: number;
  end: number;
} & (
  | {
      powerRestrictionCode: string;
      electrification: string;
    }
  | { powerRestrictionCode: typeof NO_POWER_RESTRICTION }
);
```

File: front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+        warnings.push({
+          powerRestrictionCode:
+            powerRestrictionRange.value !== NO_POWER_RESTRICTION
+              ? powerRestrictionRange.value
+              : NO_POWER_RESTRICTION,
+          electrification:
+            powerRestrictionRange.value !== NO_POWER_RESTRICTION
+              ? pathElectrificationRange.value
+              : undefined,
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts]:
```suggestion
          ...(powerRestrictionRange.value !== NO_POWER_RESTRICTION
            ? {
                powerRestrictionCode: powerRestrictionRange.value,
                electrification: pathElectrificationRange.value,
              }
            : { powerRestrictionCode: NO_POWER_RESTRICTION }),
```

**Review by Akctarus — APPROVED** [review]
LGTM, ty for the tests !

File: front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/__tests__/powerRestrictionSelector.spec.ts [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+          powerRestrictionCode: 'NO_POWER_RESTRICTION',
+          begin: 890963,
+          end: 1115651,
+        },
+      ],
+    };
+
+    expect(warnings).toEqual(expectedResult);
+  });
+  it('should return an object with 4 warnings distributed in 2 keys if warnings are a combination of one power restriction code two times and no power retriction code two times', () => {
```
Comment (clarani) [inline front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/__tests__/powerRestrictionSelector.spec.ts]:
```suggestion
  it('should return an object with 4 warnings distributed in 2 keys if warnings are a combination of one power restriction code two times and no power restriction code two times', () => {
```

**Review by clarani — CHANGES_REQUESTED** [review]
Left some comments, thanks for the PR and the tests :)

File: front/public/locales/fr/operationalStudies/manageTrainSchedule.json (hunk 1 above)
Comment (SharglutDev) [reply front/public/locales/fr/operationalStudies/manageTrainSchedule.json]: Done

File: front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx (hunk 2 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: Oh thank you for the tips !

File: front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx (hunk 3 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/PowerRestrictionsSelector.tsx]: Good point, done.

File: front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts (hunk 4 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts:6]: Very good idea, thanks

File: front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts (hunk 5 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts]: Done

File: front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/__tests__/powerRestrictionSelector.spec.ts (hunk 6 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/__tests__/powerRestrictionSelector.spec.ts]: Done

File: front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts (hunk 4 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/types.ts:6]: Actually, it adds a lot of complexity in PowerRestrictionSelector, so we decided to leave the type as it is.

File: front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts (hunk 5 above)
Comment (SharglutDev) [reply front/src/modules/trainschedule/components/ManageTrainSchedule/helpers/powerRestrictionSelector.ts]: I kept it anyway even without the type changing

**Review by clarani — APPROVED** [review]
LGTM and tested :white_check_mark:

**Pull request merged**